docs/_advanced/deferring_initialization.md

```diff
@@ -0,0 +1,30 @@
+---
+title: Deferring initialization
+lang: en
+slug: deferring-initialization
+order: 8
+---
+
+<div class="section-content">
+Bolt offers a way to defer full initialization via the `deferInitialization` option and to call the equivalent `App#init()` in your code, putting more control over asynchronous execution required for initialization into your hands as the developer.
+
+_Note: If you call `start()` before `init()`, Bolt will raise an exception._
+</div>
+
+```javascript
+const { App } = require('@slack/bolt');
+
+// deferInitialization is one of the options you can set in the constructor
+const app = new App({
+  token,
+  signingSecret,
+  deferInitialization: true,
+});
+
+// Must call init() before start() within an async function
+(async () => {
+  await app.init();
```

Comment:
Can you add try/catch clause to synchronously catch he error and terminate the process? https://github.com/slackapi/bolt-js/pull/1303/files#r799876413 the exist status should not be necessarily 255, though. `process.exist(1)` would be fine too.